tests/forking_agent/test_agent_executor.py

```diff
@@ -330,3 +332,95 @@ def test():
+    blueprint2 = executor.ExecutorBlueprint(pip_config=pip_config, requirements=requirements2, sources=sources2)
+
+    executor_manager = mpmanager_light
+    executor_1, executor_2 = await asyncio.gather(
+        executor_manager.get_executor("agent1", "local:", code_for(blueprint1), venv_checkup_interval=0.1),
+        executor_manager.get_executor("agent2", "local:", code_for(blueprint2), venv_checkup_interval=0.1),
+    )
+    executor_1_venv_status_file = pathlib.Path(executor_1.executor_virtual_env.env_path) / const.INMANTA_VENV_STATUS_FILENAME
+    executor_2_venv_status_file = pathlib.Path(executor_2.executor_virtual_env.env_path) / const.INMANTA_VENV_STATUS_FILENAME
+
+    old_datetime = datetime.datetime(year=2022, month=9, day=22, hour=12, minute=51, second=42)
+    # This part of the test is a bit subtle because we rely on the fact that there is no context switching between the
+    # modification override of the inmanta file and the retrieval of the last modification of the file
+    os.utime(
+        executor_2_venv_status_file,
+        (datetime.datetime.now().timestamp(), old_datetime.timestamp()),
+    )
+
+    old_check_executor1 = executor_1.executor_virtual_env.last_used()
+    old_check_executor2 = executor_2.executor_virtual_env.last_used()
+
+    await asyncio.sleep(0.2)
+
+    new_check_executor1 = executor_1.executor_virtual_env.last_used()
+    new_check_executor2 = executor_2.executor_virtual_env.last_used()
+
+    assert new_check_executor1 > old_check_executor1
+    assert new_check_executor2 > old_check_executor2
+    assert (datetime.datetime.now().astimezone() - new_check_executor2).seconds <= 2
+
+    # Now we want to check if the cleanup is working correctly
+    await executor_manager.stop_for_agent("agent1")
+    await asyncio.sleep(0.2)
+    # First we want to override the modification date of the `inmanta_venv_status` file
+    os.utime(
+        executor_1_venv_status_file, (datetime.datetime.now().astimezone().timestamp(), old_datetime.astimezone().timestamp())
+    )
+
+    venv_dir = pathlib.Path(mpmanager_light.environment_manager.envs_dir)
+    assert len([e for e in venv_dir.iterdir()]) == 2, "We should have two Virtual Environments for our 2 executors!"
+    # We remove the old VirtualEnvironment
+    logger.debug("Calling cleanup_virtual_environments")
+    mpmanager_light.environment_manager.running = True
+    try:
+        await asyncio.wait_for(mpmanager_light.environment_manager.cleanup_inactive_pool_members(), 0.5)
+    except TimeoutError:
+        pass
+    logger.debug("cleanup_virtual_environments ended")
+    venvs = [str(e) for e in venv_dir.iterdir()]
+    assert len(venvs) == 1, "Only one Virtual Environment should exist!"
+    assert [executor_2.executor_virtual_env.env_path] == venvs
+
+    # Let's stop the other agent and pretend that the venv is broken
+    await executor_manager.stop_for_agent("agent2")
+    await asyncio.sleep(0.2)
```

Comment:
Why this sleep?

Reply:
To make sure the `StopCommand` has been processed by the `ExecutorServer` as we don't receive a reply to confirm that everything has been stopped

Reply:
You can use `psutil` to list the subprocesses and verify that the executor process actually stopped. Sleeps like this should be prevented as much as possible. They make the test suite unstable/unpredictable and also slower than required.